### PR TITLE
chore(linting): wire up @esri/calcite-components ESLint rules

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** N/A

## Summary

Wires up custom ESLint rules for calcite-components (https://github.com/Esri/eslint-plugin-calcite-components), which includes:

* only allowing certain attributes/props to be set on <Host>
* blocking public boolean props from being initialized to false

### CODE REVIEW

File: src/components/calcite-tree/calcite-tree.tsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ export class CalciteTree {
   @Prop({ reflect: true }) theme: Theme;
 
   /** @internal If this tree is nested within another tree, set to false */
+  // eslint-disable-next-line @esri/calcite-components/strict-boolean-attributes
   @Prop({ reflect: true, mutable: true }) root = true;
```
Comment (driskull) [inline src/components/calcite-tree/calcite-tree.tsx:48]: Do we have an issue to clean this up? Even though it's internal :)

File: src/components/calcite-input-message/calcite-input-message.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ export class CalciteInputMessage {
   render(): VNode {
     const hidden = !this.active;
     return (
-      <Host calcite-hydrated-hidden={hidden} theme={this.theme}>
```
Comment (driskull) [inline src/components/calcite-input-message/calcite-input-message.tsx:66]: nice!

**Review by driskull — APPROVED** [review]
AWESOME SAUCE! 🌈 ✨

File: src/components/calcite-tree/calcite-tree.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/calcite-tree/calcite-tree.tsx:48]: I'll create one.

Comment (jcfranco) [reply src/components/calcite-tree/calcite-tree.tsx:48]: https://github.com/Esri/calcite-components/issues/2164

**Pull request merged**